Content.Server/Store/Systems/StoreSystem.Ui.cs

```diff
@@ -225,4 +306,71 @@ private void OnRequestWithdraw(EntityUid uid, StoreComponent component, StoreReq
         component.Balance[msg.Currency] -= msg.Amount;
         UpdateUserInterface(buyer, uid, component);
     }
+
+    private void OnRequestRefund(EntityUid uid, StoreComponent component, StoreRequestRefundMessage args)
+    {
+        // TODO: Remove guardian/holopara
+
+        if (args.Session.AttachedEntity is not { Valid: true } buyer)
+            return;
+
+        if (!IsOnStartingMap(uid, component))
+        {
+            component.RefundAllowed = false;
+            UpdateUserInterface(buyer, uid, component);
+        }
+
+        if (!component.RefundAllowed || !component.BoughtEntities.Any())
+            return;
+
```

Comment:
.Any() is slower for lists so you should just do count checks https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1860